### PR TITLE
Transform docker log4j properties at build time

### PR DESCRIPTION
**Pull request by pugnascotia**
For the Docker distribution, we transform the archive distribution's
log4j2 config so that all messages are logged to the console by default.
However this transformation step happens when the Docker image is built,
which means that the source for the transformation must be included in
the Docker context.

Improve this by transforming the config when the Docker context is
built. The downside is that build context task now has dependencies.

Also tweak the Checkstyle config to stop complaining about missing docs
on the `main` method.

### CODE REVIEW

File: distribution/docker/transform-log4j-config/build.gradle [hunk 1]
```diff
@@ -27,3 +24,23 @@ tasks.named('testingConventions').configure {
   }
 }
 
+String inputFile = "${parent.parent.buildDir}/outputs/default/log4j2.properties"
+String outputFile = "${project.buildDir}/distributions/log4j2.properties"
+String transformSource = "${project.projectDir}/src/main/java/org/elasticsearch/transform/log4j/TransformLog4jConfig.java"
```
Comment (mark-vieira) [inline distribution/docker/transform-log4j-config/build.gradle:29]: We should just convert `TransformLog4jConfig` into a proper Gradle task in `buildSrc` rather then rely on `LoggedExec` here. We don't use that anywhere else and it avoids forking another JVM just to execute that logic.

Comment (mark-vieira) [reply distribution/docker/transform-log4j-config/build.gradle:29]: In fact, if we do that we can ditch this project entirely and just register the transform task directly in the docker distribution project and get rid of this intermediary project altogether.

**Review by mark-vieira — CHANGES_REQUESTED** [review]

**Comment by pugnascotia** [comment]
@mark-vieira I've made the `log4j2.properties` file an artifact as you suggested, but instead of defining a task, I implemented the transformation as a filter instead. See what you think.

**Review by mark-vieira — APPROVED** [review]
Brilliant. That works just as well :+1:

**Pull request merged**